unit-tests/src/tests/test_orbital_core/unit_tests.rs

```diff
@@ -0,0 +1,286 @@
+
+    suite
+        .register_new_domain(
+            OSMOSIS_DOMAIN,
+            UncheckedOrbitalDomainConfig::Polytone {
+                note: suite.note.to_string(),
+                timeout: Uint64::zero(),
+            },
+        )
+        .unwrap();
+}
+
+#[test]
+fn test_register_orbital_domain_happy() {
+    let mut suite = OrbitalCoreBuilder::default().build();
+
+    suite
+        .register_new_domain(
+            OSMOSIS_DOMAIN,
+            UncheckedOrbitalDomainConfig::Polytone {
+                note: suite.note.to_string(),
+                timeout: Uint64::one(),
+            },
+        )
+        .unwrap();
+
+    suite
+        .register_new_domain(
+            GAIA_DOMAIN,
+            UncheckedOrbitalDomainConfig::InterchainAccount {
+                connection_id: "connection-id".to_string(),
+                channel_id: "channel-id".to_string(),
+                timeout: Uint64::one(),
+            },
+        )
+        .unwrap();
+
+    let polytone_domain = suite.query_domain(OSMOSIS_DOMAIN).unwrap();
+    let ica_domain = suite.query_domain(GAIA_DOMAIN).unwrap();
+
+    assert!(
+        polytone_domain
+            == OrbitalDomainConfig::Polytone {
+                note: suite.note,
+                timeout: Uint64::one()
+            }
+    );
+
+    assert!(
+        ica_domain
+            == OrbitalDomainConfig::InterchainAccount {
+                connection_id: "connection-id".to_string(),
+                channel_id: "channel-id".to_string(),
+                timeout: Uint64::one()
+            }
+    );
+}
+
+#[test]
+#[should_panic(expected = "User already registered")]
+fn test_register_user_duplicate() {
+    let mut suite = OrbitalCoreBuilder::default().build();
+
+    suite.register_user(USER_1).unwrap();
+    suite.register_user(USER_1).unwrap();
+}
+
+#[test]
+fn test_register_user_happy() {
+    let mut suite = OrbitalCoreBuilder::default().build();
+
+    suite.register_user(USER_1).unwrap();
+
+    let user_config = suite.query_user(USER_1).unwrap();
+
+    assert!(
+        user_config
+            == UserConfig {
+                id: Uint64::zero(),
+                registered_domains: vec![]
+            }
+    );
+}
+
+#[test]
+#[should_panic(expected = "User not registered")]
+fn test_register_user_new_domain_validates_user_registration() {
+    let mut suite = OrbitalCoreBuilder::default().build();
+
+    suite
+        .register_user_to_new_domain(USER_1, GAIA_DOMAIN, vec![])
+        .unwrap();
+}
+
+#[test]
+#[should_panic(expected = "Unknown domain: gaia")]
+fn test_register_user_new_domain_validates_domain_existance() {
+    let mut suite = OrbitalCoreBuilder::default().build();
+    suite.register_user(USER_1).unwrap();
+
+    suite
+        .register_user_to_new_domain(USER_1, GAIA_DOMAIN, vec![])
+        .unwrap();
+}
+
+#[test]
+#[should_panic(expected = "Domain registration error: no funds sent; expected 1000000untrn.")]
```

Comment:
Having amounts here hard coded is gonna be annoying if amounts will be needed to change in the future, which they often do.

Reply:
the actual amount assertions as defined on chain happen in local-interchain, this is just for confirming that the fee is being validated and returns the error